trunk/src/core/srs_core_deprecated.hpp

```diff
@@ -11,44 +11,52 @@
-// see https://github.com/ossrs/srs/discussions/3667#discussioncomment-8969107
-//
-// To free the instance in the current scope, for instance, MyClass* ptr,
-// which is a ptr and this class will:
-//       1. free the ptr.
-//       2. set ptr to NULL.
-//
-// Usage:
-//       MyClass* po = new MyClass();
-//       // ...... use po
-//       SrsAutoFree(MyClass, po);
-//
-// Usage for array:
+// See https://github.com/ossrs/srs/discussions/3667#discussioncomment-8969107 for more details.
+#define SrsAutoFree(className, instance) \
+    impl_SrsAutoFree<className> _auto_free_##instance(&instance, false, false, NULL)
+// To delete array. Please use SrsUniquePtr instead. For example:
 //      MyClass** pa = new MyClass*[size];
-//      // ....... use pa
 //      SrsAutoFreeA(MyClass*, pa);
+// This can be replaced by:
+//      SrsUniquePtr<MyClass[]> pa(new MyClass[10]);
 //
-// @remark the MyClass can be basic type, for instance, SrsAutoFreeA(char, pstr),
-//      where the char* pstr = new char[size].
-// To delete object.
-#define SrsAutoFree(className, instance) \
-    impl_SrsAutoFree<className> _auto_free_##instance(&instance, false, false, NULL)
-// To delete array.
+// Note: Please aware that there is a slight difference between SrsAutoFreeA and SrsUniquePtr. 
+// SrsAutoFreeA will track the address of pointer, while SrsUniquePtr will not.
+//      MyClass** pa;
+//      SrsAutoFreeA(MyClass*, pa); // pa will be freed even pa is changed later.
+//      SrsUniquePtr<MyClass[]> ptr(pa); // crash because pa is an invalid pointer.
+//
+// See https://github.com/ossrs/srs/discussions/3667#discussioncomment-8969107 for more details.
 #define SrsAutoFreeA(className, instance) \
     impl_SrsAutoFree<className> _auto_free_array_##instance(&instance, true, false, NULL)
-// Use free instead of delete.
-#define SrsAutoFreeF(className, instance) \
-    impl_SrsAutoFree<className> _auto_free_##instance(&instance, false, true, NULL)
-// Use hook instead of delete.
+// Use hook instead of delete. Please use SrsUniquePtr instead. For example:
+//      addrinfo* r = NULL;
+//      SrsAutoFreeH(addrinfo, r, freeaddrinfo);
+//      getaddrinfo("127.0.0.1", NULL, &hints, &r);
+// This can be replaced by:
+//      addrinfo* r = NULL;
+//      getaddrinfo("127.0.0.1", NULL, &hints, &r);
+//      SrsUniquePtr<addrinfo> ptr(r, freeaddrinfo);
+//
+// Note: Please aware that there is a slight difference between SrsAutoFreeH and SrsUniquePtr. 
+// SrsAutoFreeH will track the address of pointer, while SrsUniquePtr will not.
+//      addrinfo* r = NULL;
+//      SrsAutoFreeH(addrinfo, r, freeaddrinfo); // r will be freed even r is changed later.
+//      SrsUniquePtr<addrinfo> ptr(r, freeaddrinfo); // crash because r is an invalid pointer.
+//
+// See https://github.com/ossrs/srs/discussions/3667#discussioncomment-8969107 for more details.
 #define SrsAutoFreeH(className, instance, hook) \
```

Comment:
no used any more, delete it or not?

Reply:
I agree, so let me remove all the unused macros.